components/aws/sagemaker/tests/integration_tests/component_tests/test_batch_transform_component.py

```diff
@@ -0,0 +1,74 @@
+import utils
+import os
+import pytest
+
+from utils import kfp_client_utils
+from utils import minio_utils
+from utils import sagemaker_utils
+from utils import s3_utils
+
+
+@pytest.mark.parametrize(
+    "test_file_dir", ["resources/config/kmeans-mnist-batch-transform"]
+)
+def test_transform_job(
+    kfp_client,
+    experiment_id,
+    s3_client,
+    sagemaker_client,
+    s3_data_bucket,
+    test_file_dir,
+):
+
+    download_dir = utils.mkdir(os.path.join(test_file_dir + "/generated"))
+    test_params = utils.load_params(
+        utils.replace_placeholders(
+            os.path.join(test_file_dir, "config.yaml"),
+            os.path.join(download_dir, "config.yaml"),
+        )
+    )
+
+    # Generate random prefix for model, job name to avoid errors if resources with same name exists
+    test_params["Arguments"]["model_name"] = test_params["Arguments"][
+        "job_name"
+    ] = input_job_name = (
+        utils.generate_random_string(5) + "-" + test_params["Arguments"]["model_name"]
+    )
+
+    # Generate unique location for output since output filename is generated according to the content_type
+    test_params["Arguments"]["output_location"] = os.path.join(
+        test_params["Arguments"]["output_location"], input_job_name
+    )
+
+    run_id, status, workflow_json = kfp_client_utils.compile_run_monitor_pipeline(
```

Comment:
I might have missed it, but are these SM resources being cleaned up at the the end of the run

Reply:
Right now only endpoints are being cleaned up because they are expensive.

models and endpointconfig are not charged and hence not deleted after tests. I left them out intentionally thinking it can reduce some work while debugging

files on S3 like output models from training/hpo jobs and output from batch transform are not deleted as of now since storage is cheap, do we want to delete them?

Reply:
I think we should clean up all SM resources by default since it clutters the account. 
We can always have a flag to run a test in debug mode.

Reply:
sounds good